### PR TITLE
Add GenericModel to MOVED_IN_V2

### PR DESCRIPTION
**Pull request by adriangb**
https://github.com/pydantic/pydantic/issues/6733#issuecomment-1643724388

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by adriangb** [comment]
please review cc @Kludex

**Review by Kludex — DISMISSED** [review]

File: pydantic/_migration.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     'pydantic.utils:to_lower_camel': 'pydantic.alias_generators:to_camel',
     'pydantic:PyObject': 'pydantic.types:ImportString',
     'pydantic.types:PyObject': 'pydantic.types:ImportString',
+    'pydantic.generics.GenericModel': 'pydantic.BaseModel',
```
Comment (Kludex) [inline pydantic/_migration.py:15]: I think `GenericModel` should either on `REMOVED_IN_V2` or it should have a special condition like we have on `BaseSettings` on the `getattr_migration`. 🤔

**Review by Kludex — DISMISSED** [review]

File: pydantic/_migration.py (hunk 1 above)
Comment (adriangb) [reply pydantic/_migration.py]: Not sure what you mean, can you just push your suggestion?

**Review by dmontagu — APPROVED** [review]
Not sure what @Kludex is talking about but I'm fine with this, or with whatever other improvement over this @Kludex might have in mind.

**Comment by Kludex** [comment]
No, it was not... I was going to get to this today, I didn't have time the last days.

**Comment by adriangb** [comment]
But functionally speaking if someone replaces GenericModel with BaseModel they’re good right?

**Comment by Kludex** [comment]
> But functionally speaking if someone replaces GenericModel with BaseModel they’re good right?

Correct. 👍 

It's just that the warning is "GenericModel has been moved to BaseModel" - or similar. Anyway, it's fine.

**Pull request merged**